src/lib.rs

```diff
@@ -428,6 +436,28 @@ fn clamp_test() {
     assert!(clamp_max(::core::f32::NAN, 1.0).is_nan());
 }
 
+#[test]
+fn clamp_nan_bound() {
+    /// When debug assertions are enabled, checks that the expression panics.
+    macro_rules! assert_debug_panics {
+        ($body:expr) => {
+            if cfg!(debug_assertions) {
+                if let Ok(v) = ::std::panic::catch_unwind(|| $body) {
```

Comment:
I think you'll need to guard this test for the `std` feature, and you might as well check debug in the same guard -- `#[cfg(all(debug_assertions, feature = "std"))]`. Or you could split them into separate `#[should_panic]` tests, and not need `std`.

Note that using `cfg!` only creates a `bool`, but everything within still has to resolve.

Reply:
Oh, good catch. I've updated the PR.

Reply:
Would you prefer `#[should_panic]` tests? That would look like this:

```rust
#[test]
#[should_panic]
#[cfg(debug_assertions)]
fn clamp_nan_min() {
    clamp(0., ::core::f32::NAN, 1.);
}

#[test]
#[should_panic]
#[cfg(debug_assertions)]
fn clamp_nan_max() {
    clamp(0., -1., ::core::f32::NAN);
}

#[test]
#[should_panic]
#[cfg(debug_assertions)]
fn clamp_nan_min_max() {
    clamp(0., ::core::f32::NAN, ::core::f32::NAN);
}

#[test]
#[should_panic]
#[cfg(debug_assertions)]
fn clamp_min_nan_min() {
    clamp_min(0., ::core::f32::NAN);
}

#[test]
#[should_panic]
#[cfg(debug_assertions)]
fn clamp_max_nan_max() {
    clamp_max(0., ::core::f32::NAN);
}
```

It's longer, but the additional clarity and ability to run in `no_std` contexts is nice.

Reply:
From the CI results, it looks like using `should_panic` is the cleanest way to support Rust 1.8 (which allows attributes only on items). I've updated the PR.